### PR TITLE
chore: upgrade eslint to v8 and prettier to v2

### PR DESCRIPTION
**Pull request by huozhi**
please review this change with disabling the whitespace changes

* upgrade eslint to v7 and related packages
* upgrade prettier to v2

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -2,5 +2,7 @@
   "semi": false,
   "singleQuote": true,
   "useTabs": false,
-  "tabWidth": 2
+  "trailingComma": "none",
```
Comment (huozhi) [inline .prettierrc:5]: for compatibale with v1

File: .eslintrc [hunk 2]
```diff
@@ -46,6 +45,7 @@
     "@typescript-eslint/ban-types": 0,
     "react-hooks/rules-of-hooks": 2,
     "react-hooks/exhaustive-deps": 1,
-    "react/prop-types": 0
+    "react/prop-types": 0,
+    "testing-library/no-unnecessary-act": 0
```
Comment (huozhi) [inline .eslintrc:49]: it gives few wrong error for act

File: infinite/types.ts [hunk 3]
```diff
@@ -6,13 +6,13 @@ export type InfiniteFetcher<
   Data = any,
   KeyLoader extends InfiniteKeyLoader = InfiniteKeyLoader
 > = KeyLoader extends (...args: any[]) => any
-  ? ReturnType<KeyLoader> extends (
+  ? ReturnType<KeyLoader> extends
       | readonly [...infer K]
       | null
       | false
-      | undefined)
-    ? ((...args: [...K]) => FetcherResponse<Data>)
```
Comment (huozhi) [inline infinite/types.ts:14]: I wonder if we should keep the parentheses? @shuding wdyt

File: test/use-swr-concurrent-rendering.test.tsx [hunk 4]
```diff
@@ -29,6 +29,7 @@ describe('useSWR - concurrent rendering', () => {
 
     renderWithConfig = (element, config) =>
       act(() =>
+        // eslint-disable-next-line testing-library/no-render-in-setup
```
Comment (huozhi) [inline test/use-swr-concurrent-rendering.test.tsx:32]: it's a wrong error, we didn't call this function actually

File: test/use-swr-integration.test.tsx [hunk 5]
```diff
@@ -28,7 +28,10 @@ describe('useSWR', () => {
 
   it('should allow functions as key and reuse the cache', async () => {
     function Page() {
-      const { data } = useSWR(() => sharedKey, () => 'SWR')
+      const { data } = useSWR(
```
Comment (huozhi) [inline test/use-swr-integration.test.tsx:31]: maybe set the width to larger to keep them as before?

File: infinite/types.ts (hunk 3 above)
Comment (shuding) [reply infinite/types.ts:14]: Yeah it’s easier to read, let’s keep it!

File: test/use-swr-integration.test.tsx (hunk 5 above)
Comment (shuding) [reply test/use-swr-integration.test.tsx:31]: I think 80-char is the standard though.

**Review by shuding — APPROVED** [review]

File: infinite/types.ts (hunk 3 above)
Comment (huozhi) [reply infinite/types.ts:14]: Seems they'll be remove anyway, no options for parentheses here but only in arrow functions..

**Pull request merged**